### PR TITLE
Some code cleanup in `register_allocator.rs`

### LINKED ISSUE #1055: Improve code quality of `liveness_analysis()` in the register allocator
Clean up the implementation of https://github.com/FuelLabs/sway/blob/b1f175d36e929f7c0e3242b91df4d03d6fb4d84a/sway-core/src/asm_generation/register_allocator.rs#L53 as per Toby's comments in https://github.com/FuelLabs/sway/pull/1041

### PR DESCRIPTION
**Pull request by mohammadfawaz**
Closes #1055

### CODE REVIEW

File: sway-core/src/asm_generation/register_allocator.rs [hunk 1]
```diff
@@ -53,22 +53,17 @@ pub type InterferenceGraph =
...
-    for (ix, op) in ops.iter().enumerate() {
-        offset_to_ix.insert(op.offset, ix);
-    }
+    let offset_to_ix = HashMap::from_iter(ops.iter().enumerate().map(|(idx, op)| (op.offset, idx)));
 
-    let mut modified: bool;
-    while {
+    let mut modified = false;
+    while !modified {
         modified = false;
```
Comment (adlerjohn) [inline sway-core/src/asm_generation/register_allocator.rs:67]: By definition, if `!modified` is `true`, then `modified` is `false`.

Comment (otrho) [reply sway-core/src/asm_generation/register_allocator.rs:67]: So this line is now redundant.

But hang on, does the algorithm want to loop until nothing is newly modified, or until it's modified for the first time?

Comment (mohammadfawaz) [reply sway-core/src/asm_generation/register_allocator.rs:67]: Every outer loop needs to re-initialize `modified` to `false`. 

The algorithm needs to know if the inner loop modified anything. If it did, we need more iterations of the outer loop.

Comment (mohammadfawaz) [reply sway-core/src/asm_generation/register_allocator.rs:67]: Ok.. I updated the logic. It should be `while modified`. The first outer loop iteration has to run (like a `do while`) so I have to initialize `modifed` to `true`. Then, every outer loop iteration has to re-initialize `modified` to `false`.

Comment (otrho) [reply sway-core/src/asm_generation/register_allocator.rs:67]: Yep, that makes more sense. 👍

**Review by otrho — APPROVED** [review]

**Pull request merged**